### PR TITLE
COMMERCE-10907 Prioritize existing portletPreferences setting over default one

### PR DESCRIPTION
**Pull request by jaydawu**
https://issues.liferay.com/browse/COMMERCE-10907

Resend from https://github.com/liferay-commerce/liferay-portal/pull/3490

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceFacets.testcase [hunk 1]
```diff
@@ -14,6 +14,96 @@ definition {
...
+				key_fieldName = "$ 60.00 - $ 99.99",
+				locator1 = "FormFields#CHECKBOX");
+
+			var products = "CV Axles,Shock Absorber,Oil Pump,Bearings,Master Cylinder,Calipers,Wear Sensors,Premium Brake Fluid,Brake Fluid";
+
+			for (var i : list ${products}) {
+				AssertElementPresent(
+					key_productName = ${i},
+					locator1 = "CommerceAccelerators#PRODUCT_CARD_NAME");
+			}
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceFacets.testcase:103]: Can you rename the 'i' in these lines to more meaningful names? e.g. product or productName

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceFacets.testcase [hunk 2]
```diff
@@ -14,6 +14,96 @@ definition {
...
+			var rangesArray = '''[{'range': '[0 TO 59.99]'}\, {'range': '[60 TO 99.99]'}\, {'range': '[100 TO 199.99]'}\, {'range': '[200 TO 499.99]'}\, {'range': '[500 TO 999.99]'}\, {'range': '[1000 TO *]'}]''';
+
+			CommerceWidget.updateFacetConfiguration(
+				portletName = "Price Range Facet",
+				rangesArray = ${rangesArray});
+		}
+
+		task ("When refresh the page") {
+			Refresh();
+		}
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceFacets.testcase:34]: Is it necessary to refresh to see the updated ranges? If it is, let's keep the refresh

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]
@jaydawu please have a look at the comments. Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceFacets.testcase (hunk 2 above)
Comment (jaydawu) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceFacets.testcase:34]: @lorenzolapescara, for this bug, it was reproduced with refresh action. So if we don't use Refresh function, I think we should use another way like navigation by URL to keep covering this scenario. Which way do you think is better?

Comment (jaydawu) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceFacets.testcase:34]: @lorenzolapescara, for this bug, it was reproduced with refresh action. So if we don't use Refresh function, I think we should use another way like navigation by URL to keep covering this scenario. Which way do you think is better?

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Review by lorenzolapescara — APPROVED** [review]

**Comment by jaydawu** [comment]
ci:stop

**Comment by jaydawu** [comment]
ci:test:commerce:functional

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:forward

**Comment by jaydawu** [comment]
ci:forward

**Pull request closed without merging**